src/org/zaproxy/zap/view/widgets/WritableFileChooser.java

```diff
@@ -24,6 +24,10 @@ public WritableFileChooser(File currentDirectory) {
 	@Override
 	public void approveSelection() {
 		File selectedFile = getSelectedFile();
+		//Warns if there is less than 5MB available
+		if (selectedFile.getUsableSpace() < 5000000) {
+			JOptionPane.showMessageDialog(this, Constant.messages.getString("report.write.diskspace.warning.dialog.message"));
```

Comment:
Allowing the user to cancel the selection might be better than just warn, as it allows the user to choose other destination.